File: backend/app/schemas/market.py
from pydantic import BaseModel, Field

from app.schemas.chat import BusinessStage, RecommendationItem


class MarketInsightResponse(BaseModel):
    region: str
    industry: str
    market_summary: str
    economic_indicator: str
    consumption_trend: str
    score: float = Field(ge=0, le=100, description="상권 활성도 점수")


class PolicyFundItem(BaseModel):
    id: str
    name: str
    description: str
    eligibility: str
    max_amount: str
    stage: BusinessStage
    link: str = ""


class PolicyFundsResponse(BaseModel):
    items: list[PolicyFundItem]


class ProductRecommendRequest(BaseModel):
    region: str
    industry: str
    stage: BusinessStage
    purpose: str = ""


class ProductRecommendResponse(BaseModel):
    items: list[RecommendationItem]

File: frontend/src/types/market.ts
export interface CrisisInsight {
  level: "normal" | "warning" | "critical";
  score: number;
  summary: string;
  recommended_actions: string[];
  growth_market_names?: string[];
}

export interface MarketInsight {
  market_summary: string;
  economic_indicator: string;
  consumption_trend: string;
  crisis?: CrisisInsight | null;
  sales_data_note?: string;
}

export interface MarketInsightResponse extends MarketInsight {
  region: string;
  industry: string;
  score: number;
}

export interface PolicyFundItem {
  id: string;
  name: string;
  description: string;
  eligibility: string;
  max_amount: string;
  stage: string;
  link: string;
}

export interface RecommendationItem {
  type: "policy_fund" | "financial_product";
  name: string;
  reason: string;
  link: string;
}

export interface ChatResponse {
  answer: string;
  insights: MarketInsight | null;
  recommendations: RecommendationItem[];
  follow_up_questions: string[];
  active_agents?: string[];
}
